tests/test_fourier_correlation.py

```diff
@@ -0,0 +1,300 @@
+import logging
+import os
+import tempfile
+
+import numpy as np
+import pytest
+
+from aspire.image import Image
+from aspire.noise import BlueNoiseAdder
+from aspire.numeric import fft
+from aspire.source import Simulation
+from aspire.utils import FourierRingCorrelation, FourierShellCorrelation, Rotation
+from aspire.volume import Volume
+
+from .test_utils import matplotlib_no_gui
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "saved_test_data")
+
+logger = logging.getLogger(__name__)
+
+IMG_SIZES = [
+    64,
+    65,
+]
+DTYPES = [
+    np.float64,
+    np.float32,
+]
+METHOD = ["fft", "nufft"]
+
+
+@pytest.fixture(params=DTYPES, ids=lambda x: f"dtype={x}")
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture(params=METHOD, ids=lambda x: f"method={x}")
+def method(request):
+    return request.param
+
+
+@pytest.fixture(params=IMG_SIZES, ids=lambda x: f"img_size={x}")
+def img_size(request):
+    return request.param
+
+
+@pytest.fixture
+def image_fixture(img_size, dtype):
+    """
+    Serve up images with prescribed parameters.
+    """
+    # Load sample molecule volume
+    v = Volume(
+        np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")), dtype=dtype
+    ).downsample(img_size)
+
+    # Instantiate ASPIRE's Rotation class with a set of angles.
+    thetas = [0, 0.123]
+    rots = Rotation.about_axis("z", thetas, dtype=dtype)
+
+    # Contruct the Simulation source.
+    noisy_src = Simulation(
+        L=img_size,
+        n=2,
+        vols=v,
+        offsets=0,
+        amplitudes=1,
+        C=1,
+        angles=rots.angles,
+        noise_adder=BlueNoiseAdder.from_snr(2),
+        dtype=dtype,
+    )
+    img, img_rot = noisy_src.clean_images[:]
+    img_noisy = noisy_src.images[0]
+
+    return img, img_rot, img_noisy
+
+
+@pytest.fixture
+def volume_fixture(img_size, dtype):
+    """
+    Serve up volumes with prescribed parameters.
+    """
+    # Load sample molecule volume
+    vol = Volume(
+        np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")), dtype=dtype
+    ).downsample(img_size)
+
+    # Invert correlation for some high frequency content
+    #   Convert volume to Fourier space.
+    vol_trunc_f = fft.centered_fftn(vol.asnumpy()[0])
+    #   Get a frequency index.
+    trunc_frq = img_size // 3
+    #   Negate the power for some frequencies higher than `trunc_frq`.
```

Comment:
TBH, I was trying to avoid doing anything complicated for the tests so this didn't consume a lot of (developer) time, but I struggled to find anything simpler for the FSC (other than Identity).

You are probably right I should just use the same test pattern for images.  I'll update it.

Reply:
Replaced the slightly rotated image construction with a similar construction to volume (negates hf content).

Reply:
Thanks.